### PR TITLE
docs(actors): builtin-actors development docs

### PR DESCRIPTION
**Pull request by rvagg**
Ref: https://github.com/filecoin-project/builtin-actors/issues/1567

@rjan90 @BigLep I started banging this out as I was closing out 0081 and preparing a fresh branch based on these instructions for @snissn to get him started on MCOPY. I'll finish off the "Builtin-actors-only changes" section tomorrow, along with the MCOPY starter branch, unless something else more urgent comes up. The other sections can remain TODO until later, although the state migration one probably isn't too hard, now that I've done it for 0081; but it can be done separately.

### CODE REVIEW

**Review by BigLep — APPROVED** [review]
Guide looks great to @rvagg .  I thoroughly went through it to make sure I understood it.  I left some small suggestions.

I also added a pointer to the doc from contributing.md: https://github.com/filecoin-project/lotus/pull/12551/commits/cac1a619d2337cf896f02a58a5782aeb27fb5d09

**Comment by rvagg** [comment]
This is ready for review with the first section of builtin-actors -> lotus. But it also hopefully has enough foundational stuff that adding the additional sections for go-state-types, migrations and ref-fvm are just additive and not repetitive and tedious. I think we land this now and the remaining sections can be done over time.

File: documentation/misc/Builtin-actors_Development.md [hunk 1]
```diff
@@ -12,28 +21,34 @@ Changes to builtin-actors code can have subtle impacts on message execution, pri
...
+This method does not extend to being able to run your Lotus changes in CI with a pull request as this override is not possible in that environment. This method is only useful for local development and testing, but can be used until a tagged and released version of builtin-actors containing your changes is checked into the Lotus repository.  For a development workflow that works with CI, see [Replacing actors bundles](#replacing-actors-bundles).
 
-This method does not extend to being able to run your Lotus changes in CI with a pull request as this override is not possible in that environment. This method is only useful for local development and testing, but can be used until a tagged and released version of builtin-actors containing your changes is checked into the lotus repository.
+### Replacing actors bundles
 
-#### Replacing actors bundles
+An alternative method to the environment variable override is to simply replace the current `vX.tar.zst` file in [`build/actors/`](../../build/actors/) with your custom bundle. This method can be used to check in code for pull requests to run in CI.
```
Comment (BigLep) [inline documentation/misc/Builtin-actors_Development.md:34]: nite: feel free to disregard

```suggestion
An alternative method to the [environment variable override](#environment-variable-override) is to simply replace the current `vX.tar.zst` file in [`build/actors/`](../../build/actors/) with your custom bundle. This method can be used to check in code for pull requests to run in CI.
```

**Review by BigLep — APPROVED** [review]
I looked at the latest changes - looks good to me - nice work!

**Review by masih — APPROVED** [review]

**Pull request merged**